src/react-components/inputs/input-with-proposed-values.jsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import  React, { useState, useEffect } from 'react';
+import Input from '../../react-components/inputs/input';
+
+
+const InputWithProposedValues = ({ data, onClick, sortFunction, displayValue, ...props }) => {
+
+    const [touched, setTouched] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+    const [results, setResults] = useState([]);
+
+    useEffect(() => {
+        setInputValue('');
+        setTouched(false);
+    }, [data]);
+
+    const onChange = (value) => {
+        setInputValue(value);
+        const dataToSet = sortFunction ? data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; }).sort(sortFunction) 
```

Comment:
I think this conditional expression is quite long. I wonder if we can add those two element of this expression into the separate const functions:
'data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; }).sort(sortFunction) ' 
and 
' data.filter((element) => { return element.name.toLowerCase().indexOf(value.toLowerCase()) > -1; });' 
? 
Thanks to it this code statement could be more readable.